src/Tests/IntegrationTests/TestConnection.cs

```diff
@@ -571,8 +571,17 @@ public void CanConnectWhenHandshakeTimeoutIsSpecified()
             }
         }
 
-        /// NOT IMPLEMENTED:
-        /// TestErrOnMaxPayloadLimit
+        [Fact]
```

Comment:
A category or other attribute describing the minimum NATS version for some of these feature tests would be nice:
```suggestion
        [Fact]
        [Trait("Category", "NATS 2.1.6+")]
```